hsuforum_actions_lib.php

```diff
@@ -0,0 +1,297 @@
+<?php
+/**
+* Forum Actions
+*
+* @package   hsuforum_actions
+* @copyright 2014 Moodle Pty Ltd (http://moodle.com)
+* @author    Mikhail Janowski <mikhail@getsmarter.co.za>
+* @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+*/
+
+function hsuforum_populate_post_actions(&$posts)
+{
+   global $DB;
+   global $USER;
+
+   //Generate string for 'in' clause of query and add needed variebles to each post
+   $postidarray = array();
+   foreach ($posts as $key => $value) {
+       $postidarray[] = $key;
+
+       //Initialise post fields which get filled later
+       $posts[$key]->liked = false;//has the current user liked this post
+       $posts[$key]->likecount = 0;//how many users (exc. current user) liked this post
+       $posts[$key]->liketext = '';//text to display next to like button
+       $posts[$key]->likebuttontext = '';//text to display on the like button
+       $posts[$key]->likeaction = '';//like button action, either like or unlike
+       $posts[$key]->likeusers = array();//all users who liked this post
+
+       $posts[$key]->thanksed = false;
+       $posts[$key]->thankscount = 0;
+       $posts[$key]->thankstext = '';
+       $posts[$key]->thanksbuttontext = '';
+       $posts[$key]->thanksaction = '';
+       $posts[$key]->thanksusers = array();
+
+       $posts[$key]->actionHTML = '';
+   }
+   $postidstring = implode($postidarray, ',');
+
+
+   //Get actions for posts
+   $sql = "
+	SELECT
+	    a.id,
+	    a.postid,
+	    a.userid,
+		a.action,
+		a.created,
+		u.firstname,
+		u.lastname
+	FROM
+	    {hsuforum_actions} a
+	INNER JOIN
+	    {user} u
+	ON
+	    a.userid = u.id
+	WHERE
+	    a.postid IN($postidstring)
+	   	and action <> 'thanks'
+	";
+   $actions = $DB->get_records_sql($sql);
+
+
+   //Add actions to posts array
+   foreach ($actions as $key => $action) {
+
+       //If its a like
+       if ($action->action == 'like') {
+
+           //Add user to list
+           $user = new stdClass();
+           $user->id = $action->userid;
+           $user->time = $action->created;
+
+           if ($action->userid == $USER->id) {
+               $user->name = 'You';
+               $posts[$action->postid]->liked = true;
+           } else {
+               $user->name = $action->firstname . ' ' . $action->lastname;
+               //Add user to array
+               $posts[$action->postid]->likeusers[] = $user;
+           }
+
+           //Increment like count
+           $posts[$action->postid]->likecount++;
+
+       }
+
+       //If its a thanks
+       if ($action->action == 'thanks') {
```

Comment:
Do we still need thanks? If BSA says not we can potentially remove.

Reply:
will ask the BSA to be sure that we are removing all thanks functionality